redaxo/src/addons/structure/lib/service_article.php

```diff
@@ -54,7 +54,7 @@ public static function addArticle($data)
 
         $AART = rex_sql::factory();
         unset($id);
-        $user = rex::isBackend() ? null : 'frontend';
+        $user = rex::getUser() ? null : rex::getEnvironment();
```

Comment:
Ja klingt gut!
Und direkt in der Methode `addGlobalUpdateFields` abfangen, wenn kein User vorhanden, wäre aber verkehrt?
Praktisch wäre es ja, dann bräuchte man gar nichts mehr übergeben. Und die Methode würde selbst prüfen: user vorhanden? Dann Login-name. Ansonsten Environment.